File: src/test/java/io/github/dmitrikudrenko/core/AbstractScreen.java
package io.github.dmitrikudrenko.core;

import io.appium.java_client.AppiumDriver;
import io.appium.java_client.pagefactory.AppiumFieldDecorator;
import org.openqa.selenium.support.PageFactory;

public abstract class AbstractScreen {
    protected final AppiumDriver driver;

    public AbstractScreen(AppiumDriver driver) {
        this.driver = driver;

        PageFactory.initElements(new AppiumFieldDecorator(driver), this);
    }
}

File: src/test/java/io/github/dmitrikudrenko/core/AbstractScreen.kt
package io.github.dmitrikudrenko.core

import io.appium.java_client.AppiumDriver
import io.appium.java_client.MobileElement
import org.openqa.selenium.NoSuchElementException

abstract class AbstractScreen(private val driver: AppiumDriver<*>) {
    fun findElementById(id: String): MobileElement? {
        try {
            return driver.findElementById(id) as MobileElement?
        } catch (e: NoSuchElementException) {
            return null
        }
    }
}
